sycl/include/sycl/detail/iostream_proxy.hpp

```diff
@@ -2,7 +2,13 @@
 #include <istream>
 #include <ostream>
 
+// Hotfix to account for the different namespaces in libstdc++ and libc++
+#ifdef _LIBCPP_BEGIN_NAMESPACE_STD
+_LIBCPP_BEGIN_NAMESPACE_STD
```

Comment:
How does it play with `_MT`/`_DLL`? Or, rather, do we need anything special for libcxx on Windows?

Reply:
Also, would the following be better: https://godbolt.org/z/8x19hTfsr ? We'd need to have one .cpp file linked into the sycl.so/sycl.dll but the rest would be almost the same (std::c* -> sycl::c*).

Reply:
AFAIK the marcos doesn't touch `_MT` or `_DLL`, they just expand to slightly different namespaces. The other solution does seem to be better though, but I guess we'll need documentation to tell developers to use sycl::c* instead of std::c*

Reply:
Definitely the `sycl::detail` namespace if that's the way we want to go. This is just for internal use.